### ISSUE TITLE
Fix `function-url-quotes` false positives when URLs have modifiers

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#8545]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
	background: url( 'foo.svg' crossorigin(anonymous) );
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "function-url-quotes": "never"
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgIyTAGsBzAJwgFdMp4VKyAbAChQHIAzCCAOgGcAbiTYowFPnwhkAliWmZmSTBEwBPALZU+AShTaA3FgC+IADQgO0xjABySdXEQwAHvYAO1nmAlnwKyyQIIBiYKCjoIGSU1nwRdCFhYREc1GA40ioAtAyMmQCOlBA4MLEI4SCYMAIwZBFYYUbGvpCYAQBiUupIOEEAVpKYvrBufEEJ5Xw4qtaM8jhx5YzdJfNm9RNTM3OZLQGZk0pQSGRQCxFLxZN1mCZGQA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

No problems to be reported

### What actually happened?

The following was reported:

```
2:19-52 error Unexpected quotes around "url" function argument (function-url-quotes)
```

### Do you have a proposal to fix the bug?

The url token (`url( foo.svg )` without quotes) does not support url modifiers.
While the url function (`url( 'foo.svg' )` with quotes) does support these.

Stylelint should not report issues or try to fix url functions that contain url modifiers.

Relevant specs:
- https://www.w3.org/TR/css-values-4/#url-value
- https://drafts.csswg.org/css-values-5/#request-url-modifiers

**Comment by jeddy3** [comment]
> Stylelint should not report issues or try to fix url functions that contain url modifiers.

SGTM.

**Comment by Mouvedia** [comment]
Don't forget to support [`param()`](https://drafts.csswg.org/css-link-params/#setting-url) too.

**Comment by romainmenke** [comment]
Let's not specialize too much in this function.
Any trailer after a string token indicates that the quotes can't be removed.

We don't have to add support for specific url modifiers.
